### PR TITLE
Move legacy UnicodeEncoding tests to modern files

### PR DESCRIPTION
**Pull request by hughbe**
- Take advantage of EncodingTestHelpers and NegativeEncodingTestHelpers
to remove boilerplate and duplicated code/tests (e.g. we don't need to
test data for all overloads or GetCharCount/GetByteCount as the helpers
handle this for us)
- Move encode and decode tests to the new files

I had to add some tests:
- UnicodeEncodingEquals (e.g. Encoding.Unicode == new UnicodeEncoding(false, true))
- Convert little endian bytes to big endian bytes and make sure we can encode and decode bytes of different endianess

/cc @tarekgh

### CODE REVIEW

File: src/System.Text.Encoding/tests/UnicodeEncoding/UnicodeEncodingDecode.cs [hunk 1]
```diff
@@ -39,9 +61,45 @@ public static IEnumerable<object[]> Decode_TestData()
...
+
+        public static IEnumerable<object[]> Decode_BigEndian_TestData()
+        {
+            yield return new object[] { new byte[] { 216, 3, 48 }, 0, 3, "\uFFFD\uFFFD" };
+            yield return new object[] { new byte[] { 0, 0, 0, 84, 0, 101, 0, 10, 0, 115, 0, 116, 0, 0, 0, 9, 0, 0, 0, 84, 0, 15, 0, 101, 0, 115, 0, 116, 0, 0, 0 }, 0, 31, "\0Te\nst\0\t\0T\u000Fest\0\uFFFD" };
+
+            yield return new object[] { new byte[] { 0, 70, 0, 111, 0, 111, 0, 66, 0, 65, 4, 0, 0, 82, 70 }, 0, 15, "FooBA\u0400R\uFFFD" };
+        }
+        
+        [Theory]
```
Comment (tarekgh) [inline src/System.Text.Encoding/tests/UnicodeEncoding/UnicodeEncodingDecode.cs]: do we need to mark this method as Theory?

Comment (hughbe) [inline src/System.Text.Encoding/tests/UnicodeEncoding/UnicodeEncodingDecode.cs]: Whoops I forgot to add `[MemberData(nameof(Decode_BigEndian_TestData))]`

Comment (hughbe) [inline src/System.Text.Encoding/tests/UnicodeEncoding/UnicodeEncodingDecode.cs]: done

**Comment by tarekgh** [comment]
LGTM. by the way, fantastic job @hughbe 
I can see how the code get more organized and cleaned up.

**Pull request merged**